core/src/main/java/org/opensearch/sql/expression/DSL.java

```diff
@@ -617,145 +606,133 @@ public static WhenClause when(Expression condition, Expression result) {
+  public static FunctionExpression multi_match(Expression... args) {
     return compile(BuiltinFunctionName.MULTI_MATCH, args);
   }
 
-  public FunctionExpression simple_query_string(Expression... args) {
+  public static FunctionExpression simple_query_string(Expression... args) {
     return compile(BuiltinFunctionName.SIMPLE_QUERY_STRING, args);
   }
 
-  public FunctionExpression query(Expression... args) {
+  public static FunctionExpression query(Expression... args) {
     return compile(BuiltinFunctionName.QUERY, args);
   }
 
-  public FunctionExpression query_string(Expression... args) {
+  public static FunctionExpression query_string(Expression... args) {
     return compile(BuiltinFunctionName.QUERY_STRING, args);
   }
 
-  public FunctionExpression match_bool_prefix(Expression... args) {
+  public static FunctionExpression match_bool_prefix(Expression... args) {
     return compile(BuiltinFunctionName.MATCH_BOOL_PREFIX, args);
   }
 
-  public FunctionExpression now(Expression... args) {
+  public static FunctionExpression now(Expression... args) {
     return compile(BuiltinFunctionName.NOW, args);
   }
 
-  public FunctionExpression current_timestamp(Expression... args) {
+  public static FunctionExpression current_timestamp(Expression... args) {
     return compile(BuiltinFunctionName.CURRENT_TIMESTAMP, args);
   }
 
-  public FunctionExpression localtimestamp(Expression... args) {
+  public static FunctionExpression localtimestamp(Expression... args) {
     return compile(BuiltinFunctionName.LOCALTIMESTAMP, args);
   }
 
-  public FunctionExpression localtime(Expression... args) {
+  public static FunctionExpression localtime(Expression... args) {
     return compile(BuiltinFunctionName.LOCALTIME, args);
   }
 
-  public FunctionExpression sysdate(Expression... args) {
+  public static FunctionExpression sysdate(Expression... args) {
     return compile(BuiltinFunctionName.SYSDATE, args);
   }
 
-  public FunctionExpression curtime(Expression... args) {
+  public static FunctionExpression curtime(Expression... args) {
     return compile(BuiltinFunctionName.CURTIME, args);
   }
 
-  public FunctionExpression current_time(Expression... args) {
+  public static FunctionExpression current_time(Expression... args) {
     return compile(BuiltinFunctionName.CURRENT_TIME, args);
   }
 
-  public FunctionExpression curdate(Expression... args) {
+  public static FunctionExpression curdate(Expression... args) {
     return compile(BuiltinFunctionName.CURDATE, args);
   }
 
-  public FunctionExpression current_date(Expression... args) {
+  public static FunctionExpression current_date(Expression... args) {
     return compile(BuiltinFunctionName.CURRENT_DATE, args);
   }
 
-  private FunctionExpression compile(BuiltinFunctionName bfn, Expression... args) {
-    return (FunctionExpression) repository.compile(bfn.getName(), Arrays.asList(args.clone()));
+  @SuppressWarnings("unchecked")
```

Comment:
What is this for?

Reply:
This is to suppress the warning for type cast from `FunctionImplementation` to `T`. 

Specifically, `T` is a generic type that represents any subtype extends `FunctionImplementation`. In this way, caller don't need to convert `FunctionImplementation` to specific type again, for example, `Aggregator` or `WindowFunction`.